### PR TITLE
add windows suspend/resume hooks as @char101 sugested for others

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
Suspending is working, however, so far resume doesn't seem to work.

File: lib/Basics/process-utils.cpp [hunk 1]
```diff
@@ -1445,7 +1454,17 @@ bool TRI_SuspendExternalProcess(ExternalId pid) {
 #ifndef _WIN32
   return 0 == kill(pid._pid, SIGSTOP);
 #else
-  return true;
+  TRI_ERRORBUF;
+  
+  HANDLE processHandle = OpenProcess(PROCESS_ALL_ACCESS, FALSE, pid._pid);
+  bool rc = pfnNtSuspendProcess(processHandle) == 0;
```
Comment (mpoeter) [inline lib/Basics/process-utils.cpp]: Use `PROCESS_SUSPEND_RESUME` instead of `PROCESS_ALL_ACCESS`.
Also, you should probably check that `pfnNtSuspendProcess` is not null and that the returned handle is valid (i.e., `!= NULL`). When running under a debugger `CloseHandle` throws an exception if it is called with an invalid handle.

**Comment by dothebart** [comment]
Currently the only use cases for this code is that we have cluster resilience tests, where we pause db servers. The expectation was, that this would fix executing these tests on windows , however - it doesn't :-( 
Any idea what else we could try, or whats really wrong about this aside of unhandled edge cases?

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-linux-matrix-nightly/336/
http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-mac-matrix-nightly/179/
http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-windows-matrix-nightly/300/

File: arangod/Replication/ReplicationApplier.cpp [hunk 2]
```diff
@@ -48,12 +48,13 @@ struct ApplierThread : public Thread {
     TRI_ASSERT(_syncer);
   }
 
-  ~ApplierThread() {
+  ~ApplierThread() {    
+    shutdown();
+
     {
       MUTEX_LOCKER(locker, _syncerMutex);
       _syncer.reset();
```
Comment (mpoeter) [inline arangod/Replication/ReplicationApplier.cpp]: As already discussed via Slack - the fact that a mutex is required in the destructor is per se a sign that there is a fundamental design problem: *it should never be possible for a destructor to be executed on an object that is currently being used by some other thread.*

But this has to be addressed in a separate PR.
Anyway, resetting `_syncer` only after the thread has shut down should put out the fire for now.

**Comment by mpoeter** [comment]
@KVS85 Apart from the general issue with mutexes being used in destructors, LGTM.

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-windows-matrix-nightly/304/
http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-mac-matrix-nightly/181/
http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-linux-matrix-nightly/338/

File: js/client/modules/@arangodb/testsuites/resilience.js [hunk 3]
```diff
@@ -93,6 +93,7 @@ function activeFailover (options) {
   let testCases = tu.scanTestPaths(testPaths.active_failover);
   options.activefailover = true;
   options.singles = 4;
+  options.disableMonitor = true;
```
Comment (dothebart) [inline js/client/modules/@arangodb/testsuites/resilience.js]: please clone the options object, else it will override the settings for subsequent other testsuites.
